### ISSUE TITLE
[Bug]: Netdata client installation script fails (404 error) when downloading source code from github.com

### ISSUE THREAD

**Issue by retro1tech** [netdata/netdata#16842]
### Bug description

Netdata client installation script fails (404 error) when downloading source code from github.com.

### Expected behavior

I expect https://github.com/netdata/netdata/releases/download//netdata-.tar.gz to be downloaded, staged and installed. 

### Steps to reproduce

1. Open web browser.
2. Open zsh shell. Switch to bash shell. 
3. Open https://netdata.cloud and log in.
4. I have no active nodes. Select the macOS template.
5. Copy the script to create a macOS node:
6. Paste the script into the bash shell and run it.

### Error Logs

```
bash-5.2$ curl https://get.netdata.cloud/kickstart.sh > /tmp/netdata-kickstart.sh && sh /tmp/netdata-kickstart.sh --stable-channel --claim-token 4Cy3iArV-EG3bxDVLzQE-tZ-YYdtqxS3bPoR4hhmJwva0XTxzAr9CjsQjjt_qsNscAEYBWYUk-r32qqqNuTpcmhP2GtwTF6-LsDFH4zzYSm5jMuTNUXgN9GkHXJC8-egcMykf4g --claim-rooms f6d691d9-e6ce-42d9-9821-2aba7700f1d3 --claim-url https://app.netdata.cloud 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 87954  100 87954    0     0   391k      0 --:--:-- --:--:-- --:--:--  392k

 --- Using /var/folders/tc/b7bg05ks75b6_g3hjtn4fx940000gp/T/netdata-kickstart-XXXXXXXXXX.DrtV1I9xra as a temporary directory. --- 
 --- Checking for existing installations of Netdata... --- 
 --- No existing installations of netdata found, assuming this is a fresh install. --- 
 --- Attempting to install by building locally... --- 
[/var/folders/tc/b7bg05ks75b6_g3hjtn4fx940000gp/T/netdata-kickstart-XXXXXXXXXX.DrtV1I9xra]$ /usr/bin/curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /var/folders/tc/b7bg05ks75b6_g3hjtn4fx940000gp/T/netdata-kickstart-XXXXXXXXXX.DrtV1I9xra/install-required-packages.sh https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/install-required-packages.sh 
 OK 

 --- Running downloaded script to install required build dependencies... --- 
[/var/folders/tc/b7bg05ks75b6_g3hjtn4fx940000gp/T/netdata-kickstart-XXXXXXXXXX.DrtV1I9xra]$ /usr/local/bin/bash /var/folders/tc/b7bg05ks75b6_g3hjtn4fx940000gp/T/netdata-kickstart-XXXXXXXXXX.DrtV1I9xra/install-required-packages.sh netdata 

We detected these:
Distribution    : macos
Version         : 23.2.0
Codename        : 
Package Manager : install_brew
Packages Tree   : macos
Detection Method: uname
Default Python v: 3 

Searching for distro_sdk ...
Searching for libatomic ...
Searching for autoconf_archive ...
Searching for cmake3 ...
Searching for libz_dev ...
Searching for libuuid_dev ...
Searching for libmnl_dev ...
Searching for json_c_dev ...
Searching for fts_dev ...
Searching for libyaml_dev ...
Searching for libsystemd_dev ...
Searching for libuv ...
Searching for lz4 ...
Searching for openssl ...
Searching for zstd ...
Searching for libelf ...

All required packages are already installed. Now proceed to the next step.

 OK 

[/var/folders/tc/b7bg05ks75b6_g3hjtn4fx940000gp/T/netdata-kickstart-XXXXXXXXXX.DrtV1I9xra]$ sh -c /usr/bin/curl https://github.com/netdata/netdata/releases/latest -s -L -I -o /dev/null -w '%{url_effective}' | grep -o '[^/]*$' 
Assertion failed: (advance > 0), function procline, file util.c, line 732.
sh: line 1: 24675 Done                    /usr/bin/curl https://github.com/netdata/netdata/releases/latest -s -L -I -o /dev/null -w '%{url_effective}'
     24676 Abort trap: 6           | grep -o '[^/]*$'
 FAILED 

[/var/folders/tc/b7bg05ks75b6_g3hjtn4fx940000gp/T/netdata-kickstart-XXXXXXXXXX.DrtV1I9xra]$ /usr/bin/curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /var/folders/tc/b7bg05ks75b6_g3hjtn4fx940000gp/T/netdata-kickstart-XXXXXXXXXX.DrtV1I9xra/netdata-latest.tar.gz https://github.com/netdata/netdata/releases/download//netdata-.tar.gz 
curl: (22) The requested URL returned error: 404
 FAILED 

The following non-fatal warnings or errors were encountered:

  - Command "/usr/bin/curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /var/folders/tc/b7bg05ks75b6_g3hjtn4fx940000gp/T/netdata-kickstart-XXXXXXXXXX.DrtV1I9xra/netdata-latest.tar.gz https://github.com/netdata/netdata/releases/download//netdata-.tar.gz" failed with exit code 22.

 ABORTED  Failed to download source tarball for local build. This is usually a result of a networking issue.

For community support, you can connect with us on:
  - GitHub: https://github.com/netdata/netdata/discussions
  - Discord: https://discord.gg/5ygS846fR6
  - Our community forums: https://community.netdata.cloud/
Root privileges required to run rm -rf /var/folders/tc/b7bg05ks75b6_g3hjtn4fx940000gp/T/netdata-kickstart-XXXXXXXXXX.DrtV1I9xra
[/Users/mark]$ sudo rm -rf /var/folders/tc/b7bg05ks75b6_g3hjtn4fx940000gp/T/netdata-kickstart-XXXXXXXXXX.DrtV1I9xra 
Password:
 OK 

bash-5.2$
```

### Desktop

1. I am using a 2018 Mac Mini. 3.6 GHz quad core Intel Core i3. 8 GB 2667 MHz DDR4 RAM. 128 GB SDD with 28 GB free. macOS Sonoma 14.2.1.
2. Open Google Chrome. I am using Version 120.0.6099.216 (Official Build) (x86_64).

**Comment by hugovalente-pm** [comment]
@netdata/agent-sre something you can help check?

**Comment by ilyam8** [comment]
`grep` issue on macOS https://forums.developer.apple.com/forums/thread/738862

**Comment by ilyam8** [comment]
@retro1tech `grep` is fixed in Sonoma 14.3, update macOS.
